### ISSUE TITLE
linter: import/default flags when importing default type from external package

### ISSUE THREAD

**Issue by sbd2015** [oxc-project/oxc#23731]
### What version of Oxlint are you using?

1.60.0

### What command did you run?

oxlint --type-aware --type-check

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": [
    "typescript",
    "import"
  ],
  "options": {
    "typeAware": true,
    "typeCheck": true
  },
  "categories": {
    "correctness": "error"
  },
  "rules": {},
  "settings": {
    "import/resolver": {
      "typescript": true
    }
  },
  "env": {
    "builtin": true
  }
}
```

### What happened?

Hi,

I use oxlint integrated in vite plus originally but I still seem to have the same issue using oxlint alone.

I have an external ESM package that exports interface as default
```
export default interface SearchSpecification extends PageSpecification {
   ...
}
```

In my app, I'm importing this type as such
```
import type SearchSpecification from 'external/package/path/SearchSpecification';
```

Then [import/default rule](https://oxc.rs/docs/guide/usage/linter/rules/import/default) flags even though this type is available. I think the linter fails to see the type declaration (.d.ts).
 _× eslint-plugin-import(default): No default export found in imported module ..._

This import/default rule does not flag anymore when I switch to named import.

I hoped type-aware linting would help solve the issue so I installed oxlint-tsgolint and typescript 7.0+ [as instructed in the doc](https://oxc.rs/docs/guide/usage/linter/type-aware.html) but it didn't help.
